README.md

```diff
@@ -0,0 +1,129 @@
+Resource Allocator
+==================
+
+This project aims to help with taking care of (rather expensive)
+resources, for example several ephemeral virtual machines for the purposes
+of your CI.
+
+Overview
+--------
+
+Server side allows you to:
+  - automatically allocate resources
+  - periodically check that the resources are working properly
+  - and once the resource is not needed anymore, dispose it
+
+The client side let's your users:
+  - request particular resource type
+  - wait till the resource is available
+  - release the resource
+
+Such allocation of resource might be time consuming, so to not let your
+users wait too much -- the server side is able to pre-allocate several
+instances in advance.  For more info, have a look at `./config/pools.yaml`
+configuration example.
+
+
+What do you mean by resources?
+------------------------------
+
+Resalloc's concept of _resources_ is intentionally vague and general. Whatever
+your allocation scripts spawn and return to resalloc, those are your
+_resources_. It can be virtual machines, docker containers, IP addresses, TODO,
```

Comment:
I wanted to emphasize that resalloc doesn't necessarily have to be used only for VM spawning, so I wanted to list a couple of different resources. Do you have any good ones?

Reply:
Whatever you put here;  disk volumes, network devices, ... I think it is fine eventually. It could work even with people :-) even though the abstraction would deserve yet another iteration to better support asynchronous "human resources".